tests/unit/components/App/WorkspaceSwitcher.spec.js

```diff
@@ -1,37 +1,78 @@
 import { mount } from "@vue/test-utils";
 import WorkspaceSwitcher from "@/components/App/WorkspaceSwitcher";
+import store from "@/store";
 
 describe("WorkspaceSwitcher.vue", () => {
+  let wrapper;
+  beforeEach(() => {
+    wrapper = mount(WorkspaceSwitcher);
+  });
   it("should render with default values", () => {
-    const wrapper = mount(WorkspaceSwitcher);
     expect(wrapper).toBeTruthy();
   });
 
-  it("test updating switcher triggers the update method", () => {
-    const updateActiveWorkspace = jest.spyOn(
-      WorkspaceSwitcher.methods,
-      "updateActiveWorkspace"
-    );
-    // mock router
-    const mockRouter = {
-      push: jest.fn(),
-    };
-    const wrapper = mount(WorkspaceSwitcher, {
-      global: {
-        mocks: {
-          $router: mockRouter,
-        },
+  describe("list of workspaces provided", () => {
+    const testWorkspaces = [
+      {
+        id: 1,
+        shortcode: "test",
+        name: "TestWorkspace",
       },
+    ];
+    const testWorkspaceIndex = 0;
+    const mountWrapper = (params = {}) => {
+      store.dispatch("auth/setUser", {
+        id: 1,
+        organizations: testWorkspaces,
+      });
+      wrapper = mount(WorkspaceSwitcher, params);
+    };
+
+    beforeEach(() => {
+      mountWrapper();
+    });
+    it("displays workspaces apart from personal", () => {
+      expect(
+        wrapper.find(`[data-test="workspace-${testWorkspaceIndex + 1}"]`).text()
```

Comment:
Why + 1?
In the code also you've added the workspace ka data-test as `index  + 1`
Haven't we been keeping everything in the code as 0-indexed?